File: app/src/main/java/com/jay/currencyconverter/di/NbuActivityComponent.java
package com.jay.currencyconverter.di;

import com.jay.currencyconverter.ui.nbuActivity.INbuView;
import com.jay.currencyconverter.ui.nbuActivity.NbuActivity;

import dagger.BindsInstance;
import dagger.Component;

@Component(modules = {NbuActivityModule.class}, dependencies = {BaseComponent.class})
public interface NbuActivityComponent {

    @Component.Builder
    interface Builder {
        Builder baseComponent(BaseComponent baseComponent);
        @BindsInstance
        Builder nbuPresenter(INbuView view);
        NbuActivityComponent build();
    }

    void inject(NbuActivity activity);
}

File: app/src/main/java/com/jay/currencyconverter/di/NbuActivityComponent.kt
package com.jay.currencyconverter.di

import androidx.fragment.app.FragmentActivity
import com.jay.currencyconverter.databinding.ActivityNbuBinding
import com.jay.currencyconverter.ui.NbuActivity
import dagger.BindsInstance
import dagger.Component

@Component(modules = [NbuActivityModule::class])
interface NbuActivityComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun activity(activity: FragmentActivity): Builder
        fun build(): NbuActivityComponent
    }

    fun inject(activity: NbuActivity)
}